web/packages/teleterm/src/ui/Search/pickers/ActionPicker.tsx

```diff
@@ -125,12 +126,11 @@ export function ActionPicker(props: { input: ReactElement }) {
     }
   }
 
-  const NoResultsComponent = (
-    <NoResults
-      clusters={clustersService.getRootClusters()}
-      inputValue={inputValue}
-    />
-  );
+  const ExtraComponent = inputValue
+    ? attempts.every(a => isEmptySearch(a) && hasFinished(a)) && (
+        <NoResultsItem clusters={clustersService.getRootClusters()} />
+      )
+    : attempts.every(a => isEmptySearch(a)) && <TypeToSearchItem />;
```

Comment:
I feel like `attempts` should be exported as separate values and not an array. For the `TypeToSearchItem` case it doesn't make sense to check the resource search attempt as we know that it'll be empty. I guess this explains why there's no status check here – if you waited for the attempt to finish here, then it wouldn't solve the issue with `TypeToSearchItem` appearing only after debounce.

But another way to solve it would be to check the filter search only and show the component only if data is empty.